### PR TITLE
Enhancement/update view fn

### LINKED ISSUE #21: webhook/endpoint for updating the project_name's config
So whenever builder-backend pings core-slu-service about change in config from deployment perspective for a particular project:

* we need to update our internal state of the `Config` for a particular `project_name` whenever builder-backend pings. Presently [slu/src/api/endpoints.py](https://github.com/Vernacular-ai/dialogy-template-simple-transformers/blob/main/slu/slu/src/api/endpoints.py) doesn't have such endpoint for that. so a new route and view function for that is required.

on side note, for the upcoming launch milestones, we'll be using the core-slu for every projects. therefore have a map for `project_name` : `Config` to keep track of configs.

### PR DESCRIPTION
**Pull request by greed2411**
resolves #21 .

an endpoint/webhook to hit when new updates for existing projects, or to create new project config itself.

### CODE REVIEW

File: slu/slu/src/api/endpoints.py [hunk 1]
```diff
@@ -37,7 +37,12 @@ def slu(lang: str, project_name: str):
 
     Produces a json response containing intents and entities.
     """
-    supported_languages = list(CONFIG.get_supported_langauges().keys())
+    PROJECT_CONFIG = PROJECT_CONFIG_MAP.get(project_name, None)
+
+    if PROJECT_CONFIG is None:
+        return errors.missing_project_name(project_name), 404
+
+    supported_languages = list(PROJECT_CONFIG.get_supported_langauges().keys())
```
Comment (ltbringer) [inline slu/slu/src/api/endpoints.py:45]: Too much work? why not just

```python
if lang not in PROJECT_CONFIG.get_supported_langauges():
```

**Review by ltbringer — CHANGES_REQUESTED** [review]

File: slu/slu/utils/errors.py [hunk 2]
```diff
@@ -38,3 +38,14 @@ def invalid_input(req):
...
+def missing_project_name(project_name):
+    return (
+        jsonify(
+            {
+                "message": "Project not found.",
+                "cause": f"config for project_name: {project_name} is not present."
+            }
+        ),
+        404,
+    )
```
[the remark this answers is not in the document]
Comment (greed2411) [reply slu/slu/utils/errors.py:51]: adding it.

File: slu/slu/src/api/endpoints.py (hunk 1 above)
Comment (greed2411) [reply slu/slu/src/api/endpoints.py:45]: this PR isn't making any changes to those lines. other than replacing it with `MODEL_CONFIG`

```python
    supported_languages = list(MODEL_CONFIG.get_supported_langauges().keys())

    if lang not in supported_languages:
        return errors.invalid_language(supported_languages)
```

has been there already for rejecting requests which are not adhering to the config specified `lang`.

if you want to make those changes, this is how it'd look.

```python

    if lang not in MODEL_CONFIG.get_supported_langauges().keys():
        return errors.invalid_language(MODEL_CONFIG.get_supported_langauges().keys())
```

Comment (greed2411) [reply slu/slu/src/api/endpoints.py]: fixed it to 

```python
    if lang not in MODEL_CONFIG.get_supported_langauges():
        return errors.invalid_language(lang)
```

let me know if you have different thoughts.

**Pull request merged**